src/chain/shared_state.rs

```diff
@@ -475,21 +490,64 @@ impl SectionProofChain {
             self.blocks[block_first_index..].to_vec()
         };
 
-        SectionProofChain { genesis_pk, blocks }
+        SectionProofChain {
+            genesis_key_info,
+            blocks,
+        }
     }
 }
 
-#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
 pub struct SectionKeyInfo {
-    pub prefix: Prefix<XorName>,
-    pub version: u64,
-    pub key: BlsPublicKey,
+    // Hold all the information that is signed. When switching to real BLS, SectionKeyInfo
+    // will hold the BlsPublicKey, prefix and version and will be the item to sign.
```

Comment:
sounds like this comment shall be above the `SectionkeyInfo`, not this key_info_holder member?

Reply:
In my mind this comment is about the `key_info_holder` specifically, i.e why is it that weird thing instead of having BlsKey,Version,Prefix.